### DISCUSSION TITLE
using two channel

### DISCUSSION THREAD

**Discussion by harissutanrafiq** [mrcodetastic/ESP32-HUB75-MatrixPanel-DMA#661] [Ideas]
when I using this mocohrome library https://github.com/freetronics/DMD, i have success split a data using 2 hub12

with 2 (2 row p10 module) 32x16

PIN_R1 = 12
PIN_R2 = 12
LAT_HUB_1 = 18
LAT_HUB_2 = 19

the logic it simple 
if ( y <= 31){
  LAT_HUB_1 = HIGH;
}else{
  LAT_HUB_2 = HIGH;
}

any idea how to implement in this library ..?  i try read your code but iam could not find when you set lat pin high ;(

**Comment by board707** [comment]
> PIN_R1 = 12
> PIN_R2 = 12
> LAT_HUB_1 = 18
> LAT_HUB_2 = 19

Your split connection method does not provide any advantages over connecting two rows of panels in series. In both cases, updating two rows of panels takes about twice as long as updating one. And in your case, it is even slower, since you have to switch the LAT pin.

It is much more efficient to connect PIN_R1 and PIN_R2 to different pins, while connecting the both LAT (as well  CLK and OE) to the same. In this case, you will be able to output data to the lower and upper panel rows in true parallel mode.
In my library, I update up to seven HUB12 interfaces in parallel this way.

**Reply by harissutanrafiq** [reply]
![IMG_20240729_150721](https://github.com/user-attachments/assets/46588b9c-bb2a-4a3e-ab87-06588da06cce)

**Reply by board707** [reply]
> By separating 2 channels my customers are more familiar, rather than having to reverse the panel

The handling of chain connections with the reversing of the second row is done automatically in the library code

> I have try and still noise in a position

It looks like you switching between your two LAT pins in wrong moment...

**Reply by harissutanrafiq** [reply]
> > By separating 2 channels my customers are more familiar, rather than having to reverse the panel
> 
> The handling of chain connections with the reversing of the second row is done automatically in the library code
> 
> > I have try and still noise in a position
> 
> It looks like you switching between your two LAT pins in wrong moment...

wow thanks @board707  for fast response, any idea about this ..?
or @mrcodetastic  can you help me ..?
